core/silkworm/types/block.cpp

```diff
@@ -23,25 +23,31 @@
 
 namespace silkworm {
 
+// Just for literals
+using namespace intx;
+
 evmc::bytes32 BlockHeader::hash(bool for_sealing) const {
     Bytes rlp;
     rlp::encode(rlp, *this, for_sealing);
     return bit_cast<evmc_bytes32>(keccak256(rlp));
 }
 
+//! \brief Calculates header's boundary.
+//! \return Header's boundary.
+//! \remarks When difficulty is an exact power of 2, we need to adjust the quotient by one.
 ethash::hash256 BlockHeader::boundary() const {
-    static intx::uint256 dividend{
-        intx::from_string<intx::uint256>("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")};
-
-    ethash::hash256 ret{};
+    auto dividend{intx::uint320{1} << 256};
 
-    if (difficulty > 1u) {
-        auto result{intx::bswap(dividend / difficulty)};
-        std::memcpy(ret.bytes, intx::as_bytes(result), 32);
-    } else {
-        std::memcpy(ret.bytes, intx::as_bytes(dividend), 32);
+    if (difficulty >= 1u) {
```

Comment:
Should be strictly greater than 1. Otherwise the result is 0 for `difficulty == 1`, which is wrong.

Reply:
it already is >=

Reply:
It should be `>` instead of `>=`. Currently `BlockHeader::boundary()` returns 0 if difficulty == 1 (when you convert `intx::uint320{1} << 256` to `intx::uint256`, you get 0).

Reply:
ok, i misunderstood, now it is fixed